examples/events.py

```diff
@@ -1,217 +1,123 @@
+async def on_player_donation_receive(new_received, old_received):
+    final_received_troops = old_received.received - new_received.received
+    print(f"{new_received} of {new_received.clan} just received {final_received_troops} troops.")
+
+
+@client.event
+@coc.PlayerEvents.trophies(tags=player_tags)
 async def on_player_trophy_change(old, new):
-    # print(old.trophies, new.trophies)
-    ...
+    print(f"{new} trophies changed from {old.trophies} to {new.trophies}")
+
+
+@client.event
+@coc.PlayerEvents.versus_trophies(tags=player_tags)
+async def on_player_versus_trophy_change(old, new):
+    print(f"{new} versus trophies changed from {old.trophies} to {new.trophies}")
 
 
+"""Client Events"""
+
+
+@client.event
+@coc.ClientEvents.maintenance_start()
 async def on_maintenance():
-    log.critical("maintenance start")
-
-
-async def on_maintenance_completion(start_time):
-    log.critical("maintenance finished")
-
-
-client.on_maintenance = on_maintenance
-client.on_maintenance_completion = on_maintenance_completion
-
-
-async def get_lots_of_playertags():
-    new_tags = []
-    for _ in range(3):
-        try:
-            await client.get_clan("#abc123")
-        except:
-            pass
-    async for clan in client.get_clans(c_tags):
-        print(clan)
-        new_tags.extend(n.tag for n in clan.members)
-    return new_tags
-
-
-tags = client.loop.run_until_complete(get_lots_of_playertags())
-log.info(str(tags) + ",,")
-client.add_player_updates(tags)
-
-
-async def task():
-    process = psutil.Process()
-    for i in range(500):
-        memory = process.memory_full_info().uss / 1024 ** 2
-        print(f"Memory At {i} round: {memory:.2f} MiB")
-        objgraph.show_growth()
-        objgraph.show_most_common_types()
-        # roots = objgraph.get_leaking_objects()
-        # objgraph.show_refs(roots[:3], refcounts=True, filename=f"~graphs/chain{i}.png")
-        # objgraph.show_most_common_types(objects=roots)
-        # chain = objgraph.find_backref_chain(objgraph.by_type('cell')[-1], inspect.ismodule)
-        # in_chain = lambda x, ids=set(map(id, chain)): id(x) in ids
-        # objgraph.show_backrefs(chain[-1], len(chain), filter=in_chain, filename=f"~graphs/chain{i}.png")
-        # objgraph.show_backrefs(
-        #     random.choice(objgraph.by_type("cell")),
-        #     filename=f"~graphs/chain{i}.png"
-        # )
-        # objgraph.show_chain(objgraph.find_backref_chain(
-        #     random.choice(objgraph.by_type("cell")),
-        #     objgraph.is_proper_module
-        # ),
-        #     filename=f"~graphs/chain{i}.png"
-        # )
-
-        # Wait a few seconds
-        await asyncio.sleep(120)
-
-
-loop = asyncio.get_event_loop()
-# loop.create_task(task())
+    log.info("Maintenace Started")
+
+
+@client.event
+@coc.ClientEvents.maintenance_completion()
+async def on_maintenance_completion():
+    log.info("Maintenace Ended")
+
+
 client.loop.run_forever()
```

Comment:
If this is changed to `client.run_forever` that would be nice, as it handles cleanup in there.